packages/contracts/waffle.js

```diff
@@ -0,0 +1,85 @@
+const fs = require('fs');
+const request = require('request');
+const solcjs = require('solc');
+
+require("dotenv").config();
+
+var waffleConfig = {
+  "npmPath": "../../node_modules",
+  "legacyOutput": true,
+  "compilerOptions": {
+    "evmVersion": "constantinople"
+  }
+};
+
+var getLongVersion = (shortVersion) => {
+  return new Promise((resolve, reject) => {
+    request({
+      url: 'https://ethereum.github.io/solc-bin/bin/list.json',
+      json: true
+    }, (err, res, body) => {
+      if (!err && res.statusCode == 200) {
+        var path = body.releases[shortVersion];
+        // path of form soljson-v0.5.7+commit.6da8b019.js
+        var longVersion =
+        /soljson\-([0-9a-z\.\+]+)\.js/g.exec(path).splice(1)[0];
+        resolve(longVersion);
+      }
+    }).on('error', (err) => {
+      console.error(err);
+      reject(err);
+    });
+  });
+};
+
+var selectSolc = async (cb) => {
+  if (process.env.CI || process.env.NATIVE_SOLC) {
+    // use native solc binary for fast compilation in CI
+    waffleConfig.compiler = "native";
+
+  } else {
+
+    const solcjsVersion =
+    'v'+/([0-9a-z\.\+]+)\.Emscripten\.clang/g.exec(require("solc").version()).splice(1)[0] ;
+
+    const MigrationsContract = fs.readFileSync("contracts/Migrations.sol", "utf-8");
```

Comment:
What's the purpose of this?

Reply:
regex matching. To get the version of solcjs installed, use `require.resolve` function from nodejs, 
which has a format of `0.x.x+commit.xxxx.Emscripten.clang` , however to feed it into the `waffle config`, we need to adhere to `v0.x.x+commit.xxx`, thus the cosmetic cutting